src/framework/components/screen/system.js

```diff
@@ -80,6 +80,14 @@ pc.extend(pc, function () {
         },
 
         onRemoveComponent: function (entity, component) {
+            var _removeScreenFromElements = function (entity) {
+                entity.children.forEach(function (child) {
```

Comment:
why are you using forEach instead of a for loop?

Reply:
Coz removing components is not a performance critical bit of code and it’s easier to read and maintain.

Reply:
forEach should not be universally banned but used judiciously to keep our codebase ‘nice’. :)

Reply:
Aaarg forEach! I'd change it..

Reply:
Why?

Reply:
Why do we need another _remoeScreenFromElements function?